src/adam7.rs

```diff
@@ -6,6 +6,42 @@
 /// applies to a decoded row.
 ///
 /// See also [Reader.next_interlaced_row](crate::decoder::Reader::next_interlaced_row).
+use std::cmp::min;
+
+struct PassConstants {
+    line_mul: u8,
+    line_off: u8,
+    samp_mul: u8,
+    samp_off: u8,
```

Comment:
nit: Could you consider changing `u8` to `usize`?  This would help a tiny bit in another place below - there would be no need for `as usize` in `fn expand_adam7_bits`.